extensions/amp-render/1.0/amp-render.js

```diff
@@ -41,7 +41,63 @@ const isAmpStateSrc = (src) => src && src.startsWith(AMP_STATE_URI_SCHEME);
  */
 const isAmpScriptSrc = (src) => src && src.startsWith(AMP_SCRIPT_URI_SCHEME);
 
-class AmpRender extends BaseElement {
+/**
+ * Gets the json from an "amp-state:" uri. For example, src="amp-state:json.path".
+ *
+ * TODO: this implementation is identical to one in amp-list. Move it
+ * to a common file and import it.
+ *
+ * @param {!AmpElement} element
+ * @param {string} src
+ * @return {Promise<!JsonObject>}
+ */
+const getAmpStateJson = (element, src) => {
+  return Services.bindForDocOrNull(element)
+    .then((bind) => {
+      userAssert(bind, '"amp-state:" URLs require amp-bind to be installed.');
+      const ampStatePath = src.slice(AMP_STATE_URI_SCHEME.length);
+      return bind.getStateAsync(ampStatePath).catch((err) => {
+        const stateKey = ampStatePath.split('.')[0];
+        user().error(
+          TAG,
+          `'amp-state' element with id '${stateKey}' was not found.`
+        );
+        throw err;
+      });
+    })
+    .then((json) => {
+      userAssert(
+        json !== undefined,
+        `[amp-render] No data was found at provided uri: ${src}`
+      );
+      return json;
+    });
+};
+
+/**
+ * Returns a function to fetch json from remote url, amp-state or
+ * amp-script.
+ *
+ * @param {!AmpElement} element
+ * @return {Function}
+ */
+export const getJsonFn = (element) => {
+  const src = element.getAttribute('src');
+  if (!src) {
+    // TODO(dmanek): assert that src is provided instead of silently failing below.
+    return () => {};
+  }
+  if (isAmpStateSrc(src)) {
+    return (src) => getAmpStateJson(element, src);
+  }
+  if (isAmpScriptSrc(src)) {
```

Comment:
```suggestion
  else if (isAmpScriptSrc(src)) {
```

Reply:
Don't need `else`, each of the branches return.

Reply:
I know the code will function the same. The reason to have `else` instead of a series of `ifs` is to communicate to the reader that they are mutually exclusive. Yes one can figure it out by scanning the contents of the `if`s, but IMO its easier to grok with an else than without